db/versions/0091-migration.sql

```diff
@@ -0,0 +1,166 @@
+    pop_receipt
+  FROM azure_queue_messages
+  WHERE queue_name = 'deadline-queue'
+  AND expires > now();
+
+  CREATE INDEX queue_task_deadline_idx ON queue_task_deadlines (task_id);
+  CREATE INDEX queue_task_deadline_vis_idx ON queue_task_deadlines (visible);
+
+  -- Resolved Tasks
+  -- queries: what tasks are not scheduled for given task_group_id/scheduler_id
+  -- purpose: not much since this will have short-lived data
+  CREATE TABLE queue_resolved_tasks (
+    task_group_id text not null,
+    task_id text not null,
+    scheduler_id text not null,
+    resolution text not null,
+    resolved timestamptz not null,
+    visible timestamptz not null,
+    pop_receipt uuid null
+  );
+
+  -- migrate data to resolved queue
+  INSERT INTO
+    queue_resolved_tasks (task_id, task_group_id, scheduler_id, resolution, resolved, visible, pop_receipt)
+  SELECT
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskGroupId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'schedulerId',
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'resolution',
+    inserted,
+    visible,
+    pop_receipt
+  FROM azure_queue_messages
+  WHERE queue_name = 'resolved-queue'
+  AND expires > now();
+
+  CREATE INDEX queue_resolved_task_idx ON queue_resolved_tasks (task_id);
+  CREATE INDEX queue_resolved_task_vis_idx ON queue_resolved_tasks (visible);
+
+  -- Claimed Tasks
+  -- queries: what tasks are running and waiting to be reclaimed or resolved
+  CREATE TABLE queue_claimed_tasks (
+    task_id text not null,
+    run_id integer not null,
+    task_queue_id text not null,
+    worker_group text not null,
+    worker_id text not null,
+    claimed timestamptz not null,
+    taken_until timestamptz not null,
+    visible timestamptz not null,
+    pop_receipt uuid null
+  );
+
+  -- migrate data to resolved queue
+  INSERT INTO
+    queue_claimed_tasks (task_id, run_id, task_queue_id, worker_group, worker_id, claimed, taken_until, visible, pop_receipt)
+  SELECT
+    convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'taskId',
+    CAST(convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'runId' AS INTEGER),
+    'unknown/tq',
+    'unknonwn-wg',
+    'unknonwn-wi',
+    inserted,
+    CAST(convert_from(decode(message_text, 'base64'), 'utf-8')::jsonb->>'takenUntil' AS timestamp with time zone),
+    visible,
+    pop_receipt
+  FROM azure_queue_messages
+  WHERE queue_name = 'claim-queue'
+  AND expires > now();
+
+  CREATE INDEX queue_claimed_task_idx ON queue_claimed_tasks (task_id, run_id);
+  CREATE INDEX queue_claimed_task_vis_idx ON queue_claimed_tasks (visible);
+  CREATE INDEX queue_claimed_task_queue_idx ON queue_claimed_tasks (task_queue_id, worker_group, worker_id);
+
+
+  -- Pending (scheduled)
+  -- purpose: keep the record of all tasks that were scheduled and claimed
```

Comment:
I think claimed tasks aren't in this queue? Looking at `workclaimer.js`, the rows from this table (hints) are `.remove()`d when the task is actually claimed.

The comment at the top of that file describes this table quite nicely, actually: a task appears in this table if a tsk is pending, but a non-pending task _may_ also appear in this table.

Reply:
uh, sorry.. I didn't update comments from the initial draft before sending for review! Thanks!